### PR TITLE
Amqnet 637

### PR DESCRIPTION
**Pull request by lukeabsent**

### CODE REVIEW

**Comment by Havret** [comment]
It's all green. Awesome. :)

**Comment by lukeabsent** [comment]
:-)

**Comment by michaelpearce-gain** [comment]
Can you separate out of this PR for the No-Jiras, PR for 718 and also squash the commits, with a more meaningful commit message.

File: test/Apache-NMS-AMQP-Interop-Test/NmsMessageConsumerTest.cs [hunk 1]
```diff
@@ -310,8 +313,9 @@ public void TestSharedDurableSubscription()
 
                 messageConsumer1.Close();
                 messageConsumer2.Close();
+                
                 sessionConsumer1.Unsubscribe(subscriptionName);
-                sessionConsumer2.Unsubscribe(subscriptionName);
+                // sessionConsumer2.Unsubscribe(subscriptionName); // seems that first unsubscribe removes the first subscription
```
Comment (michaelandrepearce) [inline test/Apache-NMS-AMQP-Interop-Test/NmsMessageConsumerTest.cs:318]: no need to leave commented lines of code in.

File: test/Apache-NMS-AMQP-Test/Integration/TransactionsIntegrationTest.cs [hunk 2]
```diff
@@ -997,6 +997,12 @@ public void TestSendAfterCoordinatorLinkClosedDuringTX()
 
                 testPeer.WaitForAllMatchersToComplete(2000);
 
+                // Make sure that above line 'WaitForAllMatchersToComplete' also finishes on session/producer side
+                // In order for producer.Send to not really send, transaction on this end needs to be IsTransactionFailed
+                // in AmqpProducer  if (session.IsTransacted && session.IsTransactionFailed), and sometimes producer.Send tries
+                // to send before this flag is set, which would cause actual transfer
+                Thread.Sleep(10);
```
Comment (michaelandrepearce) [inline test/Apache-NMS-AMQP-Test/Integration/TransactionsIntegrationTest.cs:1004]: This is dubious........ anything thats a pause...sounds fishy.....

File: test/Apache-NMS-AMQP-Test/TestAmqp/TestAmqpPeerRunner.cs [hunk 3]
```diff
@@ -131,7 +130,7 @@ void Pump(Stream stream)
             catch(Exception e)
             {
                 Logger.Info(e);
-                stream.Dispose();
+                // stream.Dispose();
```
Comment (michaelandrepearce) [inline test/Apache-NMS-AMQP-Test/TestAmqp/TestAmqpPeerRunner.cs:133]: commented code?

File: test/Apache-NMS-AMQP-Interop-Test/NmsMessageConsumerTest.cs (hunk 1 above)
Comment (michaelandrepearce) [reply test/Apache-NMS-AMQP-Interop-Test/NmsMessageConsumerTest.cs:318]: @lukeabsent

File: test/Apache-NMS-AMQP-Test/Integration/TransactionsIntegrationTest.cs (hunk 2 above)
Comment (michaelandrepearce) [reply test/Apache-NMS-AMQP-Test/Integration/TransactionsIntegrationTest.cs:1004]: @lukeabsent

File: test/Apache-NMS-AMQP-Test/TestAmqp/TestAmqpPeerRunner.cs (hunk 3 above)
Comment (michaelandrepearce) [reply test/Apache-NMS-AMQP-Test/TestAmqp/TestAmqpPeerRunner.cs:133]: @lukeabsent

File: test/Apache-NMS-AMQP-Test/Integration/TransactionsIntegrationTest.cs (hunk 2 above)
Comment (lukeabsent) [reply test/Apache-NMS-AMQP-Test/Integration/TransactionsIntegrationTest.cs:1004]: I dont see much options here, the way I see it for test to go expected way, that flag mentioned in comment should switch, but I dont see any nice way of checking/waiting for it here and signal : /

**Comment by michaelandrepearce** [comment]
LGTM, great stuff thanks for fixing this up +100

**Pull request merged**